tailormap-components/projects/core/src/lib/feature-form/state/form.state-helpers.ts

```diff
@@ -31,3 +23,16 @@ export const updateFeatureInArray = (features: Feature[], newFeature: Feature):
       children: feature.children ? updateFeatureInArray(feature.children, newFeature) : null,
     }));
 };
+
+export const addFeatureToParent = (features: Feature[], newFeature: Feature, parentId:string) : Feature[] =>{
+  const idx = features.findIndex(feature => feature.objectGuid === parentId);
+
+  return (idx !== -1 ?
+      [...features.slice(0, idx),{...features[idx],children: [...features[idx].children, newFeature]},...features.slice(idx + 1)]
```

Comment:
I think this line is correct but a bit hard to read. Maybe add some breaks between the different parts?